File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/MissionTask.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class MissionTask 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Format { get; set; }
    public string Suffix { get; set; }
    
    /// <summary>
    /// Gets or Sets TaskType
    /// </summary>
    public enum TaskTypeEnum
    {
        
        /// <summary>
        /// Enum PHOTOEnum for PHOTO
        /// </summary>
        PHOTOEnum = 1,
        
        /// <summary>
        /// Enum VOTEEnum for VOTE
        /// </summary>
        VOTEEnum = 2,
        
        /// <summary>
        /// Enum ASKEnum for ASK
        /// </summary>
        ASKEnum = 3,
        
        /// <summary>
        /// Enum ANSWEREnum for ANSWER
        /// </summary>
        ANSWEREnum = 4,
        
        /// <summary>
        /// Enum GROUPEnum for GROUP
        /// </summary>
        GROUPEnum = 5,
        
        /// <summary>
        /// Enum INVITEEnum for INVITE
        /// </summary>
        INVITEEnum = 6,
        
        /// <summary>
        /// Enum OFFEREnum for OFFER
        /// </summary>
        OFFEREnum = 7,
        
        /// <summary>
        /// Enum ADEnum for AD
        /// </summary>
        ADEnum = 8,
        
        /// <summary>
        /// Enum CUSTOMEnum for CUSTOM
        /// </summary>
        CUSTOMEnum = 9,
        
        /// <summary>
        /// Enum GAMEEnum for GAME
        /// </summary>
        GAMEEnum = 10
    }

    public TaskTypeEnum TaskType { get; set; }
}

File: server-scala-play-server/app/model/MissionTask.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for MissionTask.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class MissionTask(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  name: Option[String],
  description: Option[String],
  format: Option[String],
  suffix: Option[String],
  taskType: Option[MissionTask.TaskType.Value]
)

object MissionTask {
  implicit lazy val missionTaskJsonFormat: Format[MissionTask] = Json.format[MissionTask]

  // noinspection TypeAnnotation
  object TaskType extends Enumeration {
    val PHOTO = Value("PHOTO")
    val VOTE = Value("VOTE")
    val ASK = Value("ASK")
    val ANSWER = Value("ANSWER")
    val GROUP = Value("GROUP")
    val INVITE = Value("INVITE")
    val OFFER = Value("OFFER")
    val AD = Value("AD")
    val CUSTOM = Value("CUSTOM")
    val GAME = Value("GAME")

    type TaskType = Value
    implicit lazy val TaskTypeJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

